### DISCUSSION TITLE
pprint  and sqldf

### DISCUSSION THREAD

**Discussion by gassechen** [Lisp-Stat/lisp-stat#24] [General]
Hello, I have a question.
If I use pprint to print the data frame these are the results
```
(pprint df2)
;; ESN       ASSET_ID          
;; 0-4513034   672061 
;; 2-3211115   609473 
```
but if I use them with sqldf they are these
```
(pprint (sqldf:sqldf  "select esn,ASSET_ID from DF2 limit 10"))
;; ESN       ASSET-ID
;; 0-4513034   672061
;; 2-3211115   609473
```
It is changing the names of the keys 
` ASSET_ID --> ASSET-ID`

**Comment by snunez1** [comment]
This is is supposed to be a 'feature', turning the identifiers into idiomatic lisp column names.  It's done primarily to avoid problems with SQL, that can't handle `-` in identifiers.

I can see where you might want to preserve the identifiers if you're sending the data to/from an external system, and this should probably be an enhancement.  It wouldn't be too difficult to filter on disallowed SQL characters.

What I do is mentally make this adjustment: everything that enters the system will be in 'standard' lisp format, i.e. uppercase and using `-` as a separator.

**Comment by gassechen** [comment]
Thanks Lisp-stat!!
[https://youtu.be/CX6v9fsTeno?si=78X3lLLzOLt_r4oT](url)
